hacspec/src/lib.rs

```diff
@@ -607,59 +609,122 @@ pub fn construct_state(
 }
 
 fn parse_message_1(
-    rcvd_message_1: &BytesMessage1,
-) -> (U8, BytesSupportedSuites, BytesP256ElemLen, U8) {
-    let method = rcvd_message_1[0];
-    // FIXME as we only support a fixed-sized incoming message_1,
-    // we parse directly the selected cipher suite
-    let selected_suite = BytesSupportedSuites::from_slice(rcvd_message_1, 1, 1);
-    let g_x = BytesP256ElemLen::from_slice(rcvd_message_1, 4, P256_ELEM_LEN);
-    let c_i = rcvd_message_1[MESSAGE_1_LEN - 1];
-
-    (method, selected_suite, g_x, c_i)
+    rcvd_message_1: &BufferMessage1,
+) -> Result<(U8, BytesSupportedSuites, BytesP256ElemLen, U8), EDHOCError> {
+    let mut error: EDHOCError = EDHOCError::UnknownError;
+    let method = rcvd_message_1.content[0];
+    let mut g_x: BytesP256ElemLen = BytesP256ElemLen::new();
+    let mut proposed_suites: BytesSuites = BytesSuites::from_public_array([0xff; SUITES_LEN]);
+    let mut proposed_suites_len: usize = 0;
+    let mut selected_suites: BytesSupportedSuites = BytesSupportedSuites::new();
+    let mut c_i = U8(0);
+
+    let result = match (rcvd_message_1.content[1] as U8).declassify() {
```

Comment:
Please comment that you are matching based on first byte of `SUITES_I`, which can be either an `int` or an array